Odata-docs/webapi-8/entityset-routing.md

```diff
@@ -0,0 +1,523 @@
+```
+
+For the above request to be conventionally-routed, a controller action named `Get` (or `GetShapes`) is expected, same as is expected when addressing an entity set. However, the controller action needs to be decorated with `EnableQuery` attribute. The `EnableQuery` attribute is responsible of generating the relevant query for determining the count of items:
+```csharp
+[EnableQuery]
+public ActionResult Get()
+{
+    return Ok(shapes);
+}
+```
+
+This request should return a raw value of `3`.
+
+### Addressing a collection of derived entities
+The route template for this request is: `GET ~/{entityset}/{cast}`.
+
+The following request returns a collection of derived `Rectangle` entities:
+```http
+GET http://localhost:5000/odata/Shapes/EntitySetRouting.Models.Rectangle
+```
+
+For the above request to be conventionally-routed, a controller action named `GetFromRectangle` (or `GetShapesFromRectangle`) is expected:
+```csharp
+public ActionResult GetFromRectangle()
+{
+    return Ok(shapes.OfType<Rectangle>());
+}
+```
+
+The following JSON payload shows the expected response:
+```json
+{
+    "@odata.context": "http://localhost:5000/odata/$metadata#Shapes/EntitySetRouting.Models.Rectangle",
+    "value": [
+        {
+            "Id": 1,
+            "Area": 28.0,
+            "Length": 7.0,
+            "Width": 4.0
+        },
+        {
+            "Id": 3,
+            "Area": 40.0,
+            "Length": 8.0,
+            "Width": 5.0
+        }
+    ]
+}
+```
+
+The response contains 2 rectangle objects.
+
+### Addressing the count of a collection of derived entities
+The route template for this request is: `GET ~/{entityset}/{cast}/$count`.
+
+The following request returns a count of items in the `Shapes` entity set that are rectangles:
+```http
+GET http://localhost:5000/odata/Shapes/EntitySetRouting.Models.Rectangle/$count
+```
+
+For the above request to be conventionally-routed, a controller action named `GetFromRectangle` (or `GetShapesFromRectangle`) is expected, same as is expected when addressing a collection of derived entities. However, the controller action needs to be decorated with `EnableQuery` attribute:
+```csharp
+[EnableQuery]
+public ActionResult GetFromRectangle()
+{
+    return Ok(shapes.OfType<Rectangle>());
+}
+```
+
+This request should return a raw value of `2`.
+
+### Adding an entity to an entity set
+To create an entity in an entity set, the client sends a `POST` request to that entity set's URL.
+
+The route template for this request is: `POST ~/{entityset}`
+
+The following `POST` request adds a `Shape` entity to the `Shapes` entity set:
+```http
+POST http://localhost:5000/odata/Shapes
+```
+
+Here's the request body:
+```json
+{
+    "Id": 4,
+    "Area": 36
+}
+```
+
+For the above request to be conventionally-routed, a controller action named `Post` (or `PostShape`) is expected:
+```csharp
+public ActionResult Post([FromBody] Shape shape)
+{
+    shapes.Add(shape);
+
+    return Created(shape);
```

Comment:
Is it worth mentioning why it is important to use `Created(Shape)` and not `Ok(shape)` here?

Reply:
@ElizabethOkerio I have explained that the `Created` method generates the location of the created entity and returns it as a response header